lib/executor/shared_iterations.go

```diff
@@ -212,17 +213,25 @@ func (si SharedIterations) Run(ctx context.Context, out chan<- stats.SampleConta
 		return float64(currentDoneIters) / float64(totalIters), right
 	}
 	si.progress.Modify(pb.WithProgress(progresFn))
-	go trackProgress(ctx, maxDurationCtx, regDurationCtx, &si, progresFn)
+	go trackProgress(parentCtx, maxDurationCtx, regDurationCtx, &si, progresFn)
+
+	var attemptedIters uint64
 
 	// Actually schedule the VUs and iterations...
 	activeVUs := &sync.WaitGroup{}
-	defer activeVUs.Wait()
+	defer func() {
+		activeVUs.Wait()
+		if attemptedIters < totalIters {
+			stats.PushIfNotDone(parentCtx, out, stats.Sample{
+				Value: float64(totalIters - attemptedIters), Metric: metrics.DroppedIterations,
+				Tags: si.getMetricTags(nil), Time: time.Now(),
+			})
```

Comment:
Why emit this only at the end instead of when it happens in `handleVU`?

Reply:
Because the iterations here are _shared_ between all of the VUs, we don't know how many iterations were actually dropped until _all_ of the VUs have stopped executing iterations. I just don't know what the metric value is going to be in each individual `handleVU`.

Reply:
Aha, right, some iterations might be picked up by other VUs at that point.